keps/0000-kep-template.md

```diff
@@ -139,14 +151,72 @@ What are the risks of this proposal and how do we mitigate.
 Think broadly.
 For example, consider both security and how this will impact the larger kubernetes ecosystem.
 
-## Graduation Criteria
+How will security be reviewed and by whom?
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the sig or subproject.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+What is the test plan for the component?  E2e, integration, unit testing.  How will it be tested in isolation vs
+with other components?
+
+No need to outline all of the test cases, just the general strategy.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones. These may be defined in terms of API maturity, or as something else.  Initial KEP should keep
+this high-level with a focus on what signals will be looked at to determine graduation.
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+Example 1:
+
+Alpha -> Beta Graduation:
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+
+Example 2:
+
+Beta -> GA Graduation:
+
+- N examples of real world usage
+- N installs
+- Complete features D, E, F
+
+Example 3:
+
+kubectl flag Opt-in -> Opt-out
+
+- 2 versions passed since introducing flag (to address version skew)
+- Address feedback from opt-in usage provided on GitHub issues
 
-How will we know that this has succeeded?
 Gathering user feedback is crucial for building high quality experiences and SIGs have the important responsibility of setting milestones for stability and completeness.
 Hopefully the content previously contained in [umbrella issues][] will be tracked in the `Graduation Criteria` section.
 
 [umbrella issues]: https://github.com/kubernetes/kubernetes/issues/42752
 
+### Upgrade / Downgrade Strategy
+
+**Note:** *Section not required until targeted at a release.*
```

Comment:
I'm not sure I understand the "targeted at a release" exemption... as soon as something is implementable, it generally starts to incur changes in master, which are therefore targeted at the next release.

Considering rollout during the design phase is valuable, so I'd probably drop the exemption and add some sample questions, like:
* What changes (in invocations, configurations, API use, etc) is an existing cluster required to make on upgrade in order to keep previous behavior?
* What changes (in invocations, configurations, API use, etc) is an existing cluster required to make on upgrade in order to make use of the enhancement?

Reply:
Agree

Reply:
Good point @liggitt